test/assertion/assert_unary_iteration/test_assert_unary_iteration.py

```diff
@@ -0,0 +1,261 @@
+                    "control_value": [1, 1]})
+        qc.add_gate({"name": "x", "control_qubit": [], "target_qubit": [3],
+                    "control_value": []})
+
+        expected_error_msg = "Output bitstring has an invalid length."
+
+        with self.assertRaises(QuantestPyError) as cm:
+            assert_unary_iteration(
+                circuit=qc,
+                index_reg=[0, 1],
+                system_reg=[2, 4],
+                input_to_output={"11": "010"}
+            )
+
+        self.assertEqual(cm.exception.args[0], expected_error_msg)
+
+
+class TestDrawCircuitOption(unittest.TestCase):
+
+    def setUp(self):
+        self.capture = StringIO()
+        sys.stdout = self.capture
+
+    def tearDown(self):
+        sys.stdout = sys.__stdout__
+
+    @ patch("builtins.input", return_value="")
+    def test_assert_error(self, input):
+
+        qc = QuantestPyCircuit(5)
+        qc.add_gate({"name": "x", "control_qubit": [0, 1], "target_qubit": [2],
+                    "control_value": [1, 1]})
+        qc.add_gate({"name": "z", "control_qubit": [2], "target_qubit": [3],
+                    "control_value": [1]})
+        qc.add_gate({"name": "y", "control_qubit": [2], "target_qubit": [4],
+                    "control_value": [1]})
+        qc.add_gate({"name": "x", "control_qubit": [0, 1], "target_qubit": [2],
+                    "control_value": [1, 1]})
+
+        self.assertIsNone(
+            assert_unary_iteration(
+                circuit=qc,
+                index_reg=[0, 1],
+                system_reg=[3, 4],
+                ancilla_reg=[2],
+                input_to_output={
+                    "00": "00",
+                    "01": "00",
+                    "10": "00",
+                    "11": "10"  # error
+                },
+                draw_circuit=True
+            )
+        )
+
+        stdout = self.capture.getvalue()
+        self.assertIsInstance(stdout, str)
+
+        err_msg = "In bitstring: 11\nOut bitstring expect: 10\n" \
+                  + "Out bitstring actual: 11"
+        self.assertTrue(err_msg in stdout)
+
+    @ patch("builtins.input", return_value="")
+    def test_ancilla_error(self, input):
+
+        qc = QuantestPyCircuit(5)
+        qc.add_gate({"name": "x", "control_qubit": [0, 1], "target_qubit": [2],
+                    "control_value": [1, 1]})
+        qc.add_gate({"name": "z", "control_qubit": [2], "target_qubit": [3],
+                    "control_value": [1]})
+        qc.add_gate({"name": "y", "control_qubit": [2], "target_qubit": [4],
+                    "control_value": [1]})
+        qc.add_gate({"name": "x", "control_qubit": [0, 1], "target_qubit": [2],
+                    "control_value": [0, 0]})
+
+        self.assertIsNone(
+            assert_unary_iteration(
+                circuit=qc,
+                index_reg=[0, 1],
+                ancilla_reg=[2],
+                system_reg=[3, 4],
```

Comment:
`system_reg` should be before `ancilla_reg`.

Reply:
👉  https://github.com/QuantestPy/quantestpy/pull/187/commits/af02c3fe010721ff967a38662bbb75cceeefdb76